### DISCUSSION TITLE
NAT VPS - anything special about port numbers?

### DISCUSSION THREAD

**Discussion by 4044ever** [posativ/isso#986] [Q&A]
Another beginners question. Trying to install ISSO without success on a NAT VPS with address `123.123.123.123:4231`

So far I failed. Anthing special I need to do about the port number?

```
docker pull ghcr.io/isso-comments/isso:latest

docker run -d --rm --name isso -p 127.0.0.1:8080:8080 \
    -v /var/lib/isso:/config -v /var/lib/isso:/db \
    ghcr.io/isso-comments/isso:latest
	
cd isso
	
docker-compose up -d --build
```

Build on a Debian 12 NAT VPS

**Comment by BBaoVanC** [comment]
What is the exact issue you're having; what do you mean by "So far I failed"?

**Reply by 4044ever** [reply]
Thank you for your reply.

I can build the image and start the container. But it's not showing up on the external webite.

```
# shortened a lot
[general]
name = my-external-site
dbpath = /var/lib/isso/comments.db
host =
    http://my-external-site.com
    https://my-external-site.com

[admin]
enabled = true
password = yesyes

[server]
public-endpoint = https://isso.my-external-site.com/

```

```
CONTAINER ID   IMAGE  COMMAND       CREATED         STATUS          PORTSNAMES
499fa724d4fd   isso-js-testbed   "docker-entrypoint.s…"   8 minutes ago   Exited (0) 8 minutes ago        isso-client
404ce65598cd   isso   "/isso/bin/gunicorn …"   8 minutes ago   Up 8 minutes (healthy)     127.0.0.1:8080->8080/tcp   isso-server
```

I feel like deleting all and starting again.

**Reply by BBaoVanC** [reply]
Can you look in the Network tab in the developer tools on your browser to see what response it's getting from the Isso API? Also try the admin panel to see if it works.

**Reply by 4044ever** [reply]
Seems no response from the remote site. isso.site...../admin same, no response. I think it's better to do a re-start.

Is that the correct precedure? >>

```
docker pull ghcr.io/isso-comments/isso:latest

docker run -d --rm --name isso -p 127.0.0.1:8080:8080 \
    -v /var/lib/isso:/config -v /var/lib/isso:/db \
    ghcr.io/isso-comments/isso:latest
	
cd isso     # is creates an empty isso directory?

# change isso.cfg >> /var/lib/isso/isso/isso.cfg
# change docker-compose.yml /var/lib/isso/ docker-compose.yml 
# change nginx.conf  (is that inside docker or do I change it directly on server?)
# location /var/lib/isso/contrib/nginx/nginx.conf
	
docker compose up -d --build

add script to remote site with comments
```

**isso.cfg - I added:**
host =
http://5-day-shenzhen-visa.com
https://5-day-shenzhen-visa.com

[admin]
enabled = true
password = 4711

listen = http://localhost:8080 (also tried:  :30021 and 30021:8080 )

nginx.conf +  docker-compose.yml   no change.

Update: another day of failure. Will try to get a not-NAT vps...

New Update: Left the NAT VPS, now on Oracle Cloud with a fixed IP

**Comment by ix5** [comment]
Seems you managed to fix your problem in another thread.
